design/open/pull-request.md

```diff
@@ -0,0 +1,35 @@
+# Pull-request Proposal
+
+Enable a way to submit changes to a branch by creating a request to merge changes.
+The pull-request captures the merge operation we request to apply.
+
+### Goals
+
+- Capture request to merge a branch or specific reference
+- Mechanism to review, discuss and merge changes before applying to a branch
+- Automation using actions can verify / provide feedback on pull-request
+
+### How it will work
+
+Introduce a new resource called pull-request at the repository level.
```

Comment:
What is the expected behavior if src or dst were deleted after the PR was opened?

Reply:
Good point - I'll update the proposal to act only on active branches. When you delete a branch the associated pull-requests should be closed.

Reply:
If a Branch is deleted, do we need to go over the list of PR and check if it is associated to any? Or do we close the PR only on a failed Merge attempt?
If we want to couple it, maybe we should put the PRs under the dst branch? This way If a src branch is deleted, the PR will be deleted as part of it.
Also, I think we can ignore deletion of src branch and handle the PR as an independent from it.

Reply:
We will require to close all PRs related to the branch - as a source or destination - the idea is to represent a valid request.

Failed merge will keep the PR open and notify the requester that the PR failed to merge.